spec/models/legislation/reference_spec.rb

```diff
@@ -104,4 +104,41 @@ def reference_with_type(type)
       it { is_expected.to eq false }
     end
   end
+
+  describe '#==' do
+    let(:reference) do
+      Legislation::Reference.new(legislation: foi, reference: 'Section 12(1)')
+    end
+
+    subject { reference.==(other) }
+
+    context 'other reference has matching elements' do
+      let(:other) do
+        Legislation::Reference.new(legislation: foi, reference: 'Section 12(1)')
+      end
+
+      it { is_expected.to eq true }
+    end
+
+    context 'other reference has different elements' do
+      let(:other) do
+        Legislation::Reference.new(legislation: foi, reference: 'Section 12(2)')
+      end
+      it { is_expected.to eq false }
+    end
+
+    context 'other reference has different type' do
+      let(:other) do
+        Legislation::Reference.new(legislation: foi, reference: 'Article 12(2)')
```

Comment:
This has a different "elements" as well if I understand correctly. I _think_ this should be:

```diff
- Article 12(2)
+ Article 12(1)
```